docs/smoothing.rst

```diff
@@ -29,12 +29,31 @@ will be different for each slice.
 
 Common Beam selection
 ^^^^^^^^^^^^^^^^^^^^^
-You may want to convolve your cube to the smallest beam that is still larger
-than all contained beams.  To do this, you can use the
-`~radio_beam.Beams.common_beam` tool.  For example::
 
-    common_beam = cube.beams.common_beam()
-    new_cube = cube.convolve_to(common_beam)
+Many spectral operations on a cube with a varying resolution will require first
+convolving to a common resolution. You may want to retain the finest spatial
+resolution possible by finding the smallest common beam that encloses all beams
+in the cube. To do this, we can use
+:meth:`~spectral_cube.VaryingResolutionSpectralCube.set_common_beam`::
+
+    cube.set_common_beam()
```

Comment:
Is there a reason why we need to make a stateful change on the cube and set ``common_beam``? We could instead do something like:

```
common_beam = cube.find_common_beam()
cube.convolve_to(common_beam)
```

? Having it stateful means it could get out of sync if e.g. some channels are masked and so on. In general we might want to avoid stateful changes to cubes if we can?

With ``find_common_beam`` above one could even find a common beam with different settings and compare them without having to make a stateful change each time.

Reply:
The only issue is that the common beam operation can be slow for cubes with many (>1e3) channels. And baked into the checks for deviations is the need to check for variation from the common beam, where we had the average beam before. That's our only check (currently) for whether to ignore small beam changes for operations along the spectral axis.

Allowing a stateful change could also work if we force the beam masking to not be stateful (which I think it can be now)